docs/misc/config.md

```diff
@@ -0,0 +1,95 @@
+ConfigValue<T> value = builder.comment("Comment")
+  .define("config_value_name", defaultValue);
+```
+
+The values themselves can be obtained using `ConfigValue#get`. The values are additionally cached to prevent multiple readings from files.
+
+#### Range Values
+
+Config values that are specified within a range can be defined using `#defineInRange`. These expect the value to be a subtype of `Comparable`. The method is broken down into five components: a path representing the name of the variable, the default value when no valid configuration is present, the minimum and maximum value, and a class representing the data type of the config value.
+
+#### Whitelisted Values
+
+Config values that must be a part of a specific whitelist can be defined using `#defineInList`. The method is broken down into three components: a path representing the name of the variable, the default value when no valid configuration is present, and a collection of the allowed values the configuration can be.
+
+#### List Values
+
+Config values that can take in a list of entries can be defined using `#defineList`. If the list provided by the consumer can be empty, `#defineListAllowEmpty` should be used instead. The method is broken down into three components: a path representing the name of the variable, the default value when no valid configuration is present, and a validator to make sure a deserialized element from the list is valid.
+
+#### Enum Values
+
+Config values that represent an `Enum` can be defined by `EnumValue`s via `#defineEnum`. The method is broken down into four components: a path representing the name of the variable, the default value when no valid configuration is present, a getter to convert a string or integer into an enum, and a collection of the allowed values the configuration can be.
+
+#### Boolean Values
+
+Config values that represent a `boolean` can be defined by `BooleanValue`s via `#define`. The method is broken down into two components: a path representing the name of the variable and the default value when no valid configuration is present.
+
+#### Double Values
+
+Config values that represent a `double` can be defined by `DoubleValue`s via `#defineInRange`. The method is broken down into four components: a path representing the name of the variable, the default value when no valid configuration is present, and the minimum and maximum value.
+
+#### Int Values
+
+Config values that represent an `int` can be defined by `IntValue`s via `#defineInRange`. The method is broken down into four components: a path representing the name of the variable, the default value when no valid configuration is present, and the minimum and maximum value.
+
+#### Long Values
+
+Config values that represent a `long` can be defined by `LongValue`s via `#defineInRange`. The method is broken down into four components: a path representing the name of the variable, the default value when no valid configuration is present, and the minimum and maximum value.
```

Comment:
These look very repetitive, especially the first one and last four sections. The sections on `double`, `int`, and `long` values can be defined into a single "Numerical Values" section. The information regarding the path of the config value and the default value should be deduplicated. Perhaps it may be worth converting these into lists instead of sections?